### ISSUE TITLE
[15.4.0] Can not parse in some case

### ISSUE THREAD

**Issue by yoyo837** [stylelint/stylelint#6754]
### What steps are needed to reproduce the bug?

Unable to parse when :

less file:
```less
@media screen and(width <= @screen-sm-max) {
  .topbox {
    // code
  }
}
```

Need to add spaces manually:

```less
@media screen and (width <= @screen-sm-max) {
  .topbox {
    // code
  }
}
```

### What Stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
  extends: ['stylelint-config-standard', 'stylelint-prettier/recommended'],
  overrides: [
    {
      files: ['**/*.less'],
      customSyntax: 'postcss-less',
    },
  ],
  rules: {
    'selector-pseudo-class-no-unknown': [
      true,
      {
        ignorePseudoClasses: ['global'],
      },
    ],
    'no-descending-specificity': null,
    'alpha-value-notation': null,
    'font-family-no-missing-generic-family-keyword': null,
    'declaration-empty-line-before': null,
    'selector-class-pattern': null,
    'color-function-notation': null,
    'import-notation': null,
    'length-zero-no-unit': null,
    'function-no-unknown': [
      true,
      {
        ignoreFunctions: ['env', 'constant'],
      },
    ],
  },
};

```

### How did you run Stylelint?

`.stylelintrc.js`

### Which version of Stylelint are you using?

`15.4.0`

### What did you expect to happen?

Upgrade `stylelint@15.3.0` to `15.4.0`, Lint should works.

### What actually happened?

![image](https://user-images.githubusercontent.com/6134547/229392766-29d08087-ab36-40e5-9c4b-1a4f427ffc8e.png)

### Does the bug relate to non-standard syntax?

No

### Proposal to fix the bug

Compatible with previous version.

**Comment by ybiquitous** [comment]
@yoyo837 Thanks for the report with the reproduction. We can confirm the bug also with a standard CSS code like this:

`example.css`:

```css
@media screen and(width <= @screen-sm-max) {}
```

`.stylelintrc.json`:

```json
{
  "rules": {
    "media-feature-range-notation": "context"
  }
}
```

Run:

```console
$ npx stylelint *.css

example.css
 1:8  ✖  Cannot parse media query  parseError

1 problem (1 error, 0 warnings)
```

Note: [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBbGUCWBDABAZzACcYYA7HLUqACgHcMoAXACxwB4BeHJA4sgWjwp+KLAA8AlDmABfEABoQAMwwAbGADksaBCBhjtAB3UA6MHjwLwEUioDmu4AB1yOJyEIBXdXnfxpLjhBbiBomFj8SjBYjJ7E-ISUdjD8pBCMMRg2fiGQpIz6jO6BODIucop59gBiEISijLoAVng2VrCGlojOru54jACe6qoY+TnuAIwArCYALCYADMWkcjJAA) should reproduce the problem, but it cannot. The demo app might not handle parse errors.

---

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by yoyo837** [comment]
This code has been in my project for a long time, but only stylelint@15.4.0 report an error.

**Comment by yoyo837** [comment]
I'll try to contribute later, thanks for the guidance.

**Comment by ybiquitous** [comment]
The reason that Styleint 15.4.0 fails is that we switched `postcss-media-query-parser` (outdated) to `@csstools/media-query-list-parser` (modern) for the rule's implementation. We need the latter to support the modern media query range syntax.

**Comment by romainmenke** [comment]
The error is caused by the util discussed here : https://github.com/stylelint/stylelint/issues/6559#issue-1520237343

https://github.com/stylelint/stylelint/blob/4f2ea2ef2485561c2a957c4c581ef8ad9e0bd250/lib/utils/parseMediaQuery.js#L16-L24

I think it might be better to ignore any invalid media queries here.
(removing line 16-24)

It is not possible to distinguish standard CSS syntax with a typo from non-standard syntax during/after parsing.

--------

It could be a separate rule to validate that media queries follow valid syntax, but I don't think other rules need to check and warn each time they encounter something unexpected beyond their own scope :)

**Comment by ybiquitous** [comment]
@romainmenke Thanks for the help. Your suggestion sounds good to me. 👍🏼 

Okay, let's remove `result.warn(...)` in the `parseMediaQuery()` utility.
